lib/tasks/data_import/services/import_services.rake

```diff
@@ -0,0 +1,531 @@
+            new_service_meta_exists = ServiceMeta.where(service_id: service_id)
+            if new_service_meta_exists
+              state = "updated"
+            end
+
+            new_service_meta = ServiceMeta.find_or_create_by(service_id: service_id, key: key) do |new_sm|
+              new_sm.key = key
+              new_sm.value = value
+            end
+
+            if new_service_meta
+              puts "  🟢 Service meta: #{state} (id #{new_service_meta.id})"
+            else 
+              abort("  🔴 Service meta: was not created. Exiting. #{new_service_meta.errors.messages}")
+            end
+
+          end
+        end
+      end
+    end
+  end
+
+  def import_parent_service(row)
+    if row['import_id_reference'] == nil 
+      service = Service.find_by(name: row["name"]&.strip)
+      if service
+        puts "🟠 Service: \"#{row["name"]}\" already exists, skipping anything to do with this service (id: #{service.id})."
+      else
+
+        # @TODO to/from date validation
+        # @TODO ensure booleans are booleans
+        new_service = Service.new(
+          name: row["name"]&.strip,
+          description: row["description"]&.strip,
+          url: row["url"]&.strip,
+          approved: row["approved"],
+          visible_from: row["visible_from"],
+          visible_to: row["visible_to"],
+          visible: (!row["visible_from"].nil? && !row["visible_to"].nil?) ? true : row["visible"],
+          needs_referral: row["needs_referral"],
+          free: row["free"],
+          min_age: row["min_age"],
+          max_age: row["max_age"],
+          organisation: new_service_organisation(row['organisation']),
+          temporarily_closed: row["temporarily_closed"],
+          label_list: row['labels']&.split(';')&.collect(&:strip),
+          skip_mongo_callbacks: true
+        )
+        if new_service.save
+          service_id = new_service.id
+          puts "🟢 Service: \"#{row["name"]}\" created (id: #{service_id})."
+          puts " 👉 Now adding more to it"
+
+          # deliminated data
+          # Taxonomies
+          new_service_taxonomies(new_service, row['service_taxonomies'])
+
+          # Suitabilities
+          new_service_suitabilities(new_service, row['suitabilities'])
+
+          # SEND
+          if row['is_local_offer'].present? && row['support_description'].present?
+            new_service_local_offer(service_id, row)
+          end 
+
+          # SEND needs
+          new_service_send_needs(new_service, row['send_needs_support'])
+
+
+          # custom fields
+
+          #yuck
+          # @TODO loop through CustomField.types here
+          text_rows = row.headers.filter{|h|h.starts_with?('custom_text_')}.map{|e| e.slice('custom_text_'.length, e.length) }
+          text_field_info = []
+          text_field = CustomField.where(field_type: 'text').map{|m| text_field_info << {"id" => m['id'], "key" => m['key'], "field" => m['key'].downcase.delete("^a-zA-Z0-9 ").gsub(' ', '_')}}
```

Comment:
We don't seem to be using this `text_field` variable anywhere, or any of the similar ones for the other custom field types below. Do we need them?